### DISCUSSION TITLE
Running `nvim` under a debugger; `gdb`, other?

### DISCUSSION THREAD

**Discussion by errael** [neovim/neovim#30549] [General]
Opened `termdebug` on `nvim` (running under `vim`), put a breakpoint at f_getmousepos, run... But the forking got in the way. Doing "set follow-fork-mode child" but no luck. Saw some stuff about "dap", but didn't see anyway to hook up gdb to nvim with dap.

Any guidance or suggestions on debuggering nvim?
- Any way to avoid the forking? Or how to follow the forks to get to the main nvim process where breakpoints work?
- Other recommended setups, debugger and/or front end, (other than termdebug) and best way to set up?

**Comment by zeertzjq** [comment]
This is some information in https://github.com/neovim/neovim/blob/master/src/nvim/README.md#tui-debugging-with-gdblldb
